DentOS_Framework/DentOsTestbed/src/dent_os_testbed/utils/test_utils/tb_utils.py

```diff
@@ -551,3 +552,36 @@ async def tb_device_tcpdump(device, interface, options, count_only=False, timeou
 
     else:
         return out
+
+
+async def tb_get_qualified_ports(device, ports, speed, duplex, required_ports=2):
+    """
+    Get dict of dut ports with matching speed and duplex
+    Args:
+        device (DeviceType): Dut device
+        ports (list): List of dut ports
+        speed (int): Required port speed
+        duplex (str): Required port duplex
+        required_ports (int): Required number of ports
+
+    Returns: Dictionary of qualified ports
+    """
+
+    # Check dut media mode name. If it supports the speed parameter
+    if device.media_mode == "mixed":
+        raise ValueError("Mixed mode is not supported")
+    if device.media_mode == "copper" and speed == 10000:
+        pytest.skip(reason=f"Can not run test in device {device.host_name} with the speed: {speed}")
```

Comment:
I don't think that you should skip the test inside of a utils func. This should be done inside of the test itself.

Reply:
it will be skipped if if `device.media_mode == "copper" and speed == 10000` because this is not possible technically